Update Basic Configuration (#2066)
opencast/opencast: docs/guides/admin/docs/configuration/basic.md

Basic Configuration
===================

> This guide will help you to adjust the basic configuration settings strongly recommended for each Opencast installation.

Reviewer: I'm unsure if `This guide` refers to this, the Basic Config doc, or to the OC Documentation as a whole, or ... both?

Reviewer 2: I would read `this guide` as referring to this specific page, rather than the OC docs as a whole.  Perhaps `This page` instead of `This guide` would clarify?

Reviewer: As far as I can tell, the docs use `This guide`, `This page`, and `This document` pretty interchangeably. So I suppose it can stay as is.

Author: Just explicitly named this page:

```diff
diff --git a/docs/guides/admin/docs/configuration/basic.md b/docs/guides/admin/docs/configuration/basic.md
index aa5fcf8dcf..d7821eecaf 100644
--- a/docs/guides/admin/docs/configuration/basic.md
+++ b/docs/guides/admin/docs/configuration/basic.md
@@ -1,7 +1,7 @@
 Basic Configuration
 ===================
 
-This guide will help you to adjust the basic configuration settings strongly recommended for each Opencast installation.
+The basic configuration guide will help you to adjust the settings strongly recommended for each Opencast installation.
 This is what you should do right after installing Opencast.
 There are alternatives for some of these settings but this is the recommended setup.
 

```

This is what you should do right after installing Opencast.

> There are alternatives for some of these settings but this is the recommended setup.

Reviewer: ```suggestion
While there are alternatives for some of these settings, this is the recommended setup.
```
Additionally, if there are going to be other guides for those who want to configure alternatives, that could be mentioned/linked here, a la `For those looking for guides on alternative setups, see *link*`

Author: > Additionally, if there are going to be other guides for those who want to configure alternatives

There aren't any complete guides. That information is listed on the specific modules' documentation pages and people will need to look up what they need there.

* `org.opencastproject.security.digest.user`
    * The user for the communication between Opencast nodes. It is sometimes also used by capture agents.
      This is set to `opencast_system_account` by default.
* `org.opencastproject.security.digest.pass`
    * The password for the communication between Opencast nodes. It is sometimes also used by capture agents.
      This is set to `CHANGE_ME` by default.
* `karaf.shutdown.command`
    * The security token used for shutting down Opencast. Set this to a random string.

Make sure that these settings are identical on all nodes of the cluster.

Step 3: Setting up Apache ActiveMQ Message Broker
-------------------------------------------------

Opencast requires Apache ActiveMQ to relay messages between micro-services.
For configuration details, please follow the:

- [Apache ActiveMQ configuration guide](message-broker.md)

Step 4: Database Configuration
------------------------------

Opencast uses an integrated H2 database by default, which has certain drawbacks:

* It cannot be used for distributed set-ups
* Upgrading Opencast with this database is not possible

> For testing, it is fine to keep the internal database, but configure a stand-alone database for production.

Reviewer: `For testing, it is fine to keep the internal database` -> `For testing, the internal database will suffice,`

`but configure a stand-alone database for production.` -> `but for production, a stand-alone database must/should be configured.`

Reviewer 2: Hah, so I thought about commenting on this, but decided against since it's not *technically* wrong.  I would rewrite this as
```The internal database will suffice for testing, however for production uses a stand-alone database is required```

Optionally we could include a list of databases, but I would expect anyone reading this to know what their options and preferences are.

Reviewer: You're right, it's not technically wrong x) but the OG syntax muddles the message a bit, hence the comment.

How about
`The internal database will suffice for testing, however a stand-alone database is required for production uses.`
?

Reviewer 2: That's a cleaner way of saying it, +1

Author: Went with

> The internal database will suffice for testing, however a stand-alone database is required for production uses.

Details about the configuration can be found at:

- [Database Configuration](database.md)

Step 5: Elasticsearch Configuration
-----------------------------------

Opencast requires Elasticsearch for quick full-text search and data caching.
For more details about the setup, have a look at the:

- [Elasticsearch configuration guide](../modules/searchindex/elasticsearch.md).

Step 6: HTTPS Configuration
---------------------------

> Opencast requires configuration to:

Reviewer: ```suggestion
This configuration is required in order to:
```

- Make Opencast available externally
- Secure connections from/to Opencast

For this, follow one of the

> - [configuration guides for HTTPS](https/index.md).

Reviewer: Hm... this actually links to a single page "Serve Content Via HTTPS" which then links to the three options of configuration
```suggestion
- [HTTPS configuration guides listed here](https/index.md).
```

Author: I like to keep this as it is

<!-- _Guide says nothing about how to start running OC, nor does it point to the "Next Step"
as the Installation Docs do (i.e. Install doc refers to Config doc) -->
This guide will help you to change the basic configuration settings required or at least strongly recommended for each

> Opencast installation. It is basically what you should do right after installing Opencast on your machine.  There are

Reviewer 2: `It is basically` -> `This is` ?

Author: fixed

> Opencast installation. It is basically what you should do, right after installing Opencast on your machine.  There are

Reviewer: no comma needed after "should do"

alternatives for some of these settings but this is the recommended setup.

All settings changes are made to files residing in the Opencast configuration directory.
The location of the configuration directory depends on how you installed Opencast.
If you used the Linux packages, the location is `/etc/opencast`.

Step 1: Setting the Server URL
------------------------------

> First, find the property `org.opencastproject.server.url` in `etc/custom.properties` and set your domain name.

Reviewer: suggest taking out "First", or adding "Next/Then/After that" to later steps. otherwise this feels out of place.

> The value should be the URL from which the server can be access later.

Reviewer: suggest "This value will be the URL..." instead

Author: The setting will not influence the location but must be set to the location. Maybe this is better?

```suggestion
The value must be set to the URL from which the server can be accessed later.
```

Reviewer: Yes!

    * The user for the administrative account. This is set to `admin` by default.
* `org.opencastproject.security.admin.pass`
    * The password for the administrative account. This is set to `opencast` by default.
* `org.opencastproject.security.digest.user`
    * The user for the communication between Opencast nodes. It is sometimes also used by capture agents.
      This is set to `opencast_system_account` by default.
* `org.opencastproject.security.digest.pass`
    * The password for the communication between Opencast nodes. It is sometimes also used by capture agents.
      This is set to `CHANGE_ME` by default.
* `karaf.shutdown.command`
    * The security token used for shutting down Opencast. Set this to a random string.

Make sure that these settings are identical on all nodes of the cluster.

Step 3: Setting up Apache ActiveMQ Message Broker
-------------------------------------------------

Opencast requires Apache ActiveMQ to relay messages between micro-services.
For configuration details, please follow the:

- [Apache ActiveMQ configuration guide](message-broker.md)

Step 4: Database Configuration
------------------------------

> Opencast uses an integrated H2 database by default which has certain drawbacks:

Reviewer: comma after "by default"

* It cannot be used for distributed set-ups
* Upgrading Opencast with this database is not possible

For testing, it is fine to keep the internal database, but configure a stand-alone database for production.
Details about the configuration can be found at:

- [Database Configuration](database.md)

Step 5: Elasticsearch Configuration
-----------------------------------

Opencast requires Elasticsearch for quick full-text search and data caching.
For more details about the setup, have a look at the:

- [Elasticsearch configuration guide](../modules/searchindex/elasticsearch.md).

> Step 6: HTTPS Configuration

Reviewer 3: ```suggestion
Step 6: HTTPS Configuration (Add external access)
```

Author: Let's name it <del>“Reverse Proxy Configuration”. That includes both :D</del>
…isn't such a great idea after all. Let me think about that.

---------------------------

> To make Opencast available externally and securing connections to it, follow one of the

Reviewer 2: `To make Opencast available externally and securing connections to it, ` -> `To secure connections to Opencast`

Reviewer 3: `To make Opencast available externally ` This part should be even more clear, there was an adopter with this trouble because the guide didn't clear that to have external connections, YOU NEED* to configure HTTPS, otherwise, only localhost can access. 

\* Yes you can configure `org.ops4j.pax.web.cfg` but is not recommended, That should go in a troubleshooting guide or something like that.

Reviewer 2: You don't *need* to configure https, it's sane to provide access via http-only.  It's likely not something that most adopters will want to do, but we shouldn't make it sound like https is required when it's not.

Author: For the sake of clarity, this guide should provide one way forward, not multiple ways. We should assume that new users do not know what they want (otherwise, they wouldn't read this guide). Providing options like “you can do this but if you don't do it, you need to do something else instead which is something you wouldn't need to do if you did the first thing” just makes the instructions way more confusing and will leave people in a state of “what the hell do I really need to do”.

The first paragraph of this guide states that there are alternatives to what is described in here but this is the recommended set-up. Users should not skip any step except if it is explicitly marked as optional.

I'll try to make this sentence more clear though.

EDIT: And remembering Studio right now, I think parts of Opencast will not even work any longer in a non-local HTTP set-up. I think the web video stuff needs HTTPS iirc.
